controllers/web.js

```diff
@@ -0,0 +1,193 @@
+            }
+          });
+        })
+        .catch( (err) => {
+          res.logger.error(err);
+          return sendResponse(res, "Oups : " + err.message);
+        });
+
+      }
+    });
+
+
+  }
+
+
+  function getHistory(req, res) {
+    let senderId = req.params.senderId;
+    if (!senderId) {
+      return res.status(200).end();
+    } else {
+      return web.getHistory(res, senderId);
+    }
+  }
+
+
+
+
+  function onGet(req, res) {
+    //Il faut un id unique à chaque utilisateur
+
+    let senderId = req.query.senderId;
+    res.senderId = senderId;
+    if (senderId) {
+      return web.getUnread(res,senderId);
+    } else {
+      //nouvelle connection
+      senderId = uuid();
+      let but = [
+        new Button("web_url", `${config.server.url}${config.server.basePath}/authorize?state=${senderId}-web`,"se connecter")
+      ];
+      sendResponse(senderId, new ButtonsListMessage("Pour commencer, connecter vous :)", but));
+    }
+    return res.status(200).send(senderId);
+  }
+
+
+  function onPost(req, res) {
+    let message;
+    let senderId;
+    let type;
+
+    message = req.body.message;
+    senderId = req.body.senderId;
+    type = req.body.type;
+
+    res.senderId = senderId;
+
+    //save user message to db
+    //formatMsg
+    let userMsg = {
+      message,
+      origin: "user"
+    };
+    WebMessage.findOneAndUpdate({senderId}, {$push: {history: userMsg}}, {upsert: true}) //TODO implement history deletion
+      .exec()
+      .catch(err => {
+        res.logger.error(err);
+      });
+
+
+    if (type === "message") {
+      return messageReceived(res, senderId, message);
+    } else if(type === "postback") {
+      return postbackReceived(res, senderId, message);
+    }
+
+
+
+  }
+
+  function sendQuickResponses(res, senderId, responses) {
+    return Bluebird.mapSeries(responses, (response) => {
+      switch (response.type) {
+      case 0:
+      default:
+        let textResponse = response.speech.replace(/<(.*)\|+(.*)>/, "$1");
+        return sendResponse(res, senderId, textResponse);
+      }
+    });
+  }
+
+  function sendFeedback(res, senderId, intent, message) {
+    if (intent === "unknown") {
+      return;
+    }
+
+    message = message.length >= 1000 ? "TOOLONG" : message;
+
+    let buttons = [
+      new Button("postback", `FEEDBACK_MISUNDERSTOOD_${camelCase(intent)}_${message}`, "Mauvaise compréhension"),
+      new Button("postback", `FEEDBACK_BAD_${camelCase(intent)}_${message}`, "Non"),
+      new Button("postback", `FEEDBACK_GOOD_${camelCase(intent)}_${message}`, "Oui")
+    ];
+
+    return sendResponse(res, senderId, new ButtonsListMessage("Est-ce que cette réponse vous a aidé ?", buttons));
+
+  }
+
+  function sendResponse(res, senderId, response) {
+    if (!response) {
+      response = senderId;
+      senderId = res;
+    }
+
+    return new Bluebird( (resolve) => {
+      return resolve(web.send(res || senderId, response));
```

Comment:
Nope, just `return web.send(res || senderId, response);` because it seems it's already a promise. But please check if you always return a promise from `web.send(...)`